### PR TITLE
Entity reference import

### PR DESCRIPTION
**Pull request by TheSlimvReal**
closes: #1768 

### Visible/Frontend Changes
- [x] Entity properties can be selected and allow to reference other entities through a matching property

### Architectural/Backend Changes
--

### CODE REVIEW

File: src/app/core/entity/schema-datatypes/array.datatype.spec.ts [hunk 1]
```diff
@@ -19,18 +17,12 @@ describe("Schema data type: array", () => {
...
-        ComponentRegistry,
-        {
-          provide: DefaultDatatype,
-          useValue: new ConfigurableEnumDatatype({
-            getEnumValues: () => defaultInteractionTypes,
-          } as any),
-          multi: true,
-        },
-      ],
+      imports: [MockedTestingModule.withState()],
```
Comment (sleidig) [inline src/app/core/entity/schema-datatypes/array.datatype.spec.ts:20]: in the spirit of "unit tests" trying to minimize the external dependency of the test, I had intentionally put up this test setup. But if the MockedTestingModule helps avoid further problems, that's also cool.

File: src/app/features/import/import-column-mapping/entity-value-mapping/entity-value-mapping.component.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import { FormBuilder, FormControl, ReactiveFormsModule } from "@angular/forms";
+import { MatFormFieldModule } from "@angular/material/form-field";
+import { MatSelectModule } from "@angular/material/select";
+import { NgForOf } from "@angular/common";
+import { MatButtonModule } from "@angular/material/button";
+import { EntityConstructor } from "../../../../core/entity/model/entity";
+import { HelpButtonComponent } from "../../../../core/common-components/help-button/help-button.component";
+import { DynamicComponent } from "../../../../core/view/dynamic-components/dynamic-component.decorator";
+
+@DynamicComponent("EntityValueMapping")
```
Comment (sleidig) [inline src/app/features/import/import-column-mapping/entity-value-mapping/entity-value-mapping.component.ts:19]:
```suggestion
/**
 * Configuration UI for the EntityDatatype's import mapping function.
 */
@DynamicComponent("EntityValueMapping")
```

File: src/app/features/import/import-column-mapping/entity-value-mapping/entity-value-mapping.component.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      .map(([prop, schema]) => ({ label: schema.label, property: prop }));
+    this.propertyForm.setValue(this.data.col.additional);
+  }
+
+  async save() {
+    const confirmed =
+      this.propertyForm.value ||
+      (await this.confirmation.getConfirmation(
+        $localize`Cancel?`,
+        $localize`If no property is selected, no import will be done.`,
```
Comment (sleidig) [inline src/app/features/import/import-column-mapping/entity-value-mapping/entity-value-mapping.component.ts:61]:
```suggestion
        $localize`Ignore for import?`,
        $localize`If no property is selected, this column will be skipped during import. Are you sure to keep it unmapped?`,
```

**Review by sleidig — APPROVED** [review]
Looks very clean and works well! :smiley:

File: src/app/core/entity/schema-datatypes/array.datatype.spec.ts (hunk 1 above)
Comment (TheSlimvReal) [reply src/app/core/entity/schema-datatypes/array.datatype.spec.ts:20]: Yes, unfortunately to whole datatype dependencies would require a lot of mocking

**Pull request merged**